pop/POPAnimator.mm

```diff
@@ -347,8 +366,25 @@ - (id)init
   _displayLink.paused = YES;
   [_displayLink addToRunLoop:[NSRunLoop mainRunLoop] forMode:NSRunLoopCommonModes];
 #else
-  CVDisplayLinkCreateWithActiveCGDisplays(&_displayLink);
-  CVDisplayLinkSetOutputCallback(_displayLink, displayLinkCallback, (__bridge void *)self);
+  CVReturn ret = CVDisplayLinkCreateWithActiveCGDisplays(&_displayLink);
+  if (kCVReturnSuccess != ret) {
+    ret = CVDisplayLinkCreateWithCGDisplay(CGMainDisplayID(), &_displayLink);
+  }
+  if (kCVReturnSuccess == ret) {
+    CVDisplayLinkSetOutputCallback(_displayLink, displayLinkCallback, (__bridge void *)self);
+  } else {
+    FBLogAnimInfo(@"cannot create display link: ret=%ld, falling back to display timer at %llu Hz", (long)ret, kDisplayTimerFrequency);
+    _displayTimer = dispatch_source_create(DISPATCH_SOURCE_TYPE_TIMER, 0, 0, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0));
+    NSAssert(nil != _displayTimer, @"Cannot create display timer");
+    dispatch_source_set_timer(_displayTimer, DISPATCH_TIME_NOW, NSEC_PER_SEC / kDisplayTimerFrequency, 0);
+    __weak POPAnimator *weakSelf = self;
+    dispatch_source_set_event_handler(_displayTimer, ^{
+      __strong POPAnimator *strongSelf = weakSelf;
+      if (__builtin_expect(nil != strongSelf, 1)) {
```

Comment:
Would you like to change this by calling a private method for display timer creation?